dips/dip-8.md

```diff
@@ -1,55 +1,144 @@
+3. The user approves the FPA in their wallet (VASP)
+4. The user wallet VASP hands a billing agreement id (FPPA id) to the merchant VASP
+
+Whenever the app wants to charge the user: 
+1. The merchant VASP sends a payment request to the user wallet VASP mentioning the FPA id
+2. If the payment request does not fall out of the FPA scope, the payment is continued normally as the user was approved it (without user interaction)
+
+---
+# Disclaimer
+---
+
+This DIP does not contain the phase of exchanging identifiers between VASPs that is required to negotiate merchant scenarios.
+
+For the purpose of this document, we will assume that given a payment object with two VASP addresses and some underline indetifier, there is a protocol for exchaning these identities under VASPs (buyer and merchant) without revealing their identities to the public. 
+
+The process described below starts at the phase that both sides have the relevant payment unique identifier (reference) and each VASP knows to corrlate it to its respectful entity under. 
+
+---
+# Synonyms
+---
+
+For the sake of simplicity, we use the following terms:
+* "Buyer VASP" - The VASP who submits payment transactions to the blockchain in order to pay to the other VASP
+* "Merchant" - The VASP who recieves the funds (transaction)
+* "Buyer user" - The buyer user who wants to pay to the merchant
 
 ---
 # Specification
 ---
 
-# Fund Pull Pre-Approval
-### *VERSION SUPPORT: Supported only in version 1 of off-chain APIs*
+### *VERSION SUPPORT: Supported only in version 3 of off-chain APIs*
+
+Establishes a relationship between a buyer VASP and a merchant where the merchant can then pull funds from the buyer VASP without the buyer user approving each transaction, like billing the user without an interaction for each payment. This relationship marked by a unique identifier, that is the `**funds_pull_pre_approval_id**`. 
+
+## Funds pull pre approval request fields
+
+The following fields should be used to define a funds pull pre approval request. These fields should be encoded into a series of URL parameters appended to the query string. In case of QR code, the image should include the full link with all parameters. 
+
+| Field                  | Type    | Required?  | Description                                    |
+|------------------------|---------|------------|------------------------------------------------|
+| vasp_address               | str      | Y          | Address of account from which billing will happen. The address is encoded using bech32 and should uniquly identify the merchant. For Diem addresses format, refer to the "account identifiers" section in [DIP-5](https://dip.diem.com/dip-5/#account-identifiers). |
+| funds_pull_pre_approval_id | str      | Y          | A unique identifier of this pre-approval. Should be a UUID according to RFC4122 with "-"'s included. |
+| merchant_name              | str      | Y          | The name of the biller |
```

Comment:
I like the use of `merchant` here. Compared to DIP 158 however, our naming seems inconsistent. In DIP 158 we're hesitant to be narrow and tight in our definitions, and use "payment receiver" and the like. It would be great to be consistent.

On that note, should this object be a `PaymentReceiverObject`? Seems like no if entirely sent over URL.